packages/paste-core/components/table/src/types.ts

```diff
@@ -14,70 +16,84 @@ export interface TableContextProps {
 
 export interface TableProps extends React.TableHTMLAttributes<HTMLTableElement> {
   children: NonNullable<React.ReactNode>;
+  element?: BoxProps['element'];
```

Comment:
Is this the same as writing `Pick<BoxProps, 'element'>`?

Reply:
Not quite. `Pick<BoxProps, 'element'>` represents an object with only the `element` key picked from `BoxProps`, while `BoxProps['element']` gives you just the type of the key `element` from box props.

Reply:
Technically this could be done by extending the interface again with `Partial<Pick<BoxProps, 'element'>>`, but IMHO this is hard to read and the intellisense is confusing, since it doesn't extract the type, rather it will show the chaining of utility types.

Reply:
Kind of, you definitely could have done this and achieved the same thing, as it takes the key, the requirement and value.

```
export interface TableProps extends React.TableHTMLAttributes<HTMLTableElement>, Pick<BoxProps, 'element'> {
  children: NonNullable<React.ReactNode>;
  ...
```

`element: BoxProps['element'];` is useful if you want to change the requirement of element, from optional to required